client/sniffer/src/main/java/org/elasticsearch/client/sniff/ElasticsearchHostsSniffer.java

```diff
@@ -26,22 +26,25 @@
 import org.apache.commons.logging.LogFactory;
 import org.apache.http.HttpEntity;
 import org.apache.http.HttpHost;
+import org.elasticsearch.client.Node;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.Node.Roles;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.URI;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.concurrent.TimeUnit;
 
 /**
  * Class responsible for sniffing the http hosts from elasticsearch through the nodes info api and returning them back.
- * Compatible with elasticsearch 5.x and 2.x.
+ * Compatible with elasticsearch 2.x+.
  */
 public final class ElasticsearchHostsSniffer implements HostsSniffer {
```

Comment:
it is annoying that the name is off now, as we sniff nodes and not hosts. But I think that we want to keep things this way for backwards compatibility? Maybe we should start deprecating things and adjust the naming though? But wait this is already a breaking change as we return nodes rather than hosts, should we also  rename the interface?

Reply:
That is a very good question!

I think everything will still compile if folks update their version and they don't have any `HostsSniffer` subclasses. Maybe we merge this to PR to master and 6.x and we rename the class in master only?

I think we can change the name of the interface and users of `ElasticsearchHostsSniffer` won't be broken but folks who implement the interface will be.

We *could* implement backwards compatibility by adding a `NodesSniffer` interface but that seems a bit much at this point.

Reply:
But even if we don't rename, users who implement the interface will have to adapt their code due to the method signature change? So I think we could break this too? I don't expect many users, if any, to implement this interface at the moment.